wled00/FX_fcn.cpp

```diff
@@ -1181,18 +1181,21 @@ uint32_t Segment::color_wheel(uint8_t pos) const {
  * @returns Single color from palette
  */
 uint32_t Segment::color_from_palette(uint16_t i, bool mapping, bool wrap, uint8_t mcol, uint8_t pbri) const {
-  uint32_t color = gamma32(currentColor(mcol));
-
+  
+  uint32_t color = currentColor(mcol);
   // default palette or no RGB support on segment
-  if ((palette == 0 && mcol < NUM_COLORS) || !_isRGB) return (pbri == 255) ? color : color_fade(color, pbri, true);
+  if ((palette == 0 && mcol < NUM_COLORS) || !_isRGB) {
+    color = gamma32(color);
+    return (pbri == 255) ? color : color_fade(color, pbri, true);
+  }
 
-  unsigned paletteIndex = i;
+  uint8_t paletteIndex = i;
```

Comment:
why?

Reply:
it is only used as an uint8_t down the line and not involved in any calculation.

Reply:
yes. ATM. what about in the future?
if it is unsigned, compiler will trim it down when passing to a function.

Reply:
ok reverted to unsigned locally.

Reply:
XTensa doesn't have non-word arithmetic instructions -- any operations done on a 8-bit or 16-bit types have to be performed by upcasting, operating, and downcasting afterwards to restrict the output range.  At least here, `int` or `unsigned` is almost always less code (and slightly faster).

Reply:
> XTensa doesn't have non-word arithmetic instructions -- any operations done on a 8-bit or 16-bit types have to be performed by upcasting, operating, and downcasting afterwards to restrict the output range. At least here, `int` or `unsigned` is almost always less code (and slightly faster).

in general: absolutely yes. But if a function is performed purely in 8bit (like most of the fastled stuff) sometimes the compiler optimizes in a way that makes it actually faster (smaller code) when using `uint8_t`, don't ask me why exactly, I have been playing with types to improve code quite a bit and sometimes its just not really clear what is happening.